src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -369,8 +369,9 @@ private void enableStorage(XWalkView container) {
 	private void enableUrlHandlers(XWalkView container) {
 		container.setResourceClient(new XWalkResourceClient(container) {
 			@Override public boolean shouldOverrideUrlLoading(XWalkView view, String url) {
-				if(url.startsWith("tel:") || url.startsWith("sms:")) {
-					Intent i = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+				Uri uri = Uri.parse(url);
+				if(url.startsWith("tel:") || url.startsWith("sms:") || !isUrlRelated(appUrl, uri)) {
```

Comment:
I think we can just call `isUrlRelated` and drop the startsWith calls, eg:

```suggestion
				if(!isUrlRelated(appUrl, uri)) {
```

Then, maybe it makes sense to invert the `if` so you're asking a positive.

As a nice to have, I see that `isUrlRelated` takes a `Uri` but immediately `toStrings` it - would it be worth adding a variant which takes a string so we don't have to parse it most of the time?

Finally, we actually have a few tests for the Utils so if you do the above checks it'd be worth adding unit tests to assert that `sms` and `tel` and `mailto` links fire the intent.

Reply:
Done!